File: 351004/Brazhalovich/Lab_1/src/main/java/org/example/newsapi/mapper/NewsMapper.java
package org.example.newsapi.mapper;

import org.example.newsapi.dto.request.NewsRequestTo;
import org.example.newsapi.dto.response.NewsResponseTo;
import org.example.newsapi.entity.News;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingTarget;

@Mapper(componentModel = "spring")
public interface NewsMapper {
    @Mapping(target = "id", ignore = true)
    @Mapping(target = "created", ignore = true)
    @Mapping(target = "modified", ignore = true)
    News toEntity(NewsRequestTo request);

    NewsResponseTo toDto(News news);

    @Mapping(target = "id", ignore = true)
    @Mapping(target = "created", ignore = true)
    @Mapping(target = "modified", ignore = true)
    void updateEntityFromDto(NewsRequestTo request, @MappingTarget News news);
}

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/mapper/NewsMapper.kt
package com.distcomp.mapper

import com.distcomp.dto.news.NewsRequestTo
import com.distcomp.dto.news.NewsResponseTo
import com.distcomp.entity.News
import org.mapstruct.Mapper
import org.mapstruct.Mapping
import org.mapstruct.MappingConstants
import org.mapstruct.ReportingPolicy
import java.time.LocalDateTime

@Mapper(
    componentModel = MappingConstants.ComponentModel.SPRING,
    unmappedTargetPolicy = ReportingPolicy.IGNORE
)
interface NewsMapper {
    @Mapping(target = "userId", source = "user.id")
    fun toNewsResponse(news: News) : NewsResponseTo

    fun toNewsEntity(newsRequestTo: NewsRequestTo): News {
        return News(
            id = newsRequestTo.id,
            title = newsRequestTo.title,
            content = newsRequestTo.content
        )
    }
}